src/protocols/aeternity/helpers/transactionTokenInfoResolvers.ts

```diff
@@ -396,17 +392,15 @@ const buy: TransactionResolver = (transaction, tokens = null, tokenAddressMapper
  */
 const sell: TransactionResolver = (transaction, tokens = null, tokenAddressMapper = undefined) => {
   const isConfirm = !transaction.tx.return;
-  const tokenAddress: string = isConfirm && tokenAddressMapper
-    ? tokenAddressMapper(transaction.tx.contractId)
-    : transaction.tx.log?.slice(-1)[0]?.address;
+  const tokenAddress: string | undefined = tokenAddressMapper?.(transaction.tx.contractId);
 
   const amount = isConfirm ? transaction.tx.arguments?.[0]?.value : transaction.tx.arguments?.[0]!;
   const aeAmount = isConfirm ? transaction.tx.arguments?.[1]?.value : transaction.tx.return?.value;
   const token = {
     amount: amount?.value,
     ...defaultToken,
-    symbol: tokens?.[tokenAddress]?.symbol,
-    ...tokens?.[tokenAddress],
+    symbol: tokenAddress && tokens?.[tokenAddress]?.symbol,
```

Comment:
Not sure if `tokenAddress &&` is necessary. Wouldn't it return `undefined` either way from the `tokens?.[tokenAddress]?`? Same in line 373.